New Snackbar/Toast using Materialize CSS

Mohammed-JS-Khan opened the pull request:
Added the new Materialize CSS Toast to replace the snackbar (it stacks). Modified colours to work with the scheme (please input if you like the colours or not). Fixed #38 : Changed the one Snackbar message for visibility toggle that was the same when questions were made visible and invisible.
![image](https://user-images.githubusercontent.com/9061203/31753772-bb228dd2-b460-11e7-8a3f-c8fcd23d94b6.png)

Mohammed-JS-Khan committed:
Added new functionality to the snackbar to make it persist when spammed.

Mohammed-JS-Khan committed:
Removed an extra un-needed return statement.

Mohammed-JS-Khan committed:
Merge branch 'develop' of https://github.com/quizzard-edu/quizzard into develop

Mohammed-JS-Khan committed:
Redesigned snackbar using Materialize CSS Toasts

Mohammed-JS-Khan committed:
Merge branch 'develop' of https://github.com/quizzard-edu/quizzard into develop

ahmedqarmout2 commented on public/js/admin.js:
```diff
@@ -436,7 +436,10 @@ var updateVisibility = function(qid) {
         success: function(data) {
             displayQuestionTable();
-            dropSnack(colours.SUCCESS_GREEN, 'Question ' + qid + ' is now visible to the students');
+
```
No need for new lines here. Also both options say visible to the student. Am I missing something or shouldn't the second option say not visible?

ahmedqarmout2 replied:
I see the invisible. It would be better to say not visible, I barely noticed the (in)visible.

Mohammed-JS-Khan replied:
@ahmedqarmout2 Yessir, sounds good. What if i wrote "visible" and "not visible" and i made the former a blue snack/toast and the latter a yellow snack/toast? Ill try to add an icon but I'm not sure if it would mess with the snack/toast's animations in any way.

What does everyone think?

ahmedqarmout2 replied:
Sounds great, except the blue color XD why blue? I mean if there is a meaning for blue then sure yeah, blue seems odd XD

Mohammed-JS-Khan replied:
@ahmedqarmout2 LOL cause I couldn't think of any other colour. I wanna have different colours for "visible" and "not visible" so its more apparent. Suggestions? Or should I just try for icons and call it a day?

ahmedqarmout2 replied:
Go with green instead of blue.

Mohammed-JS-Khan replied:
Same green as the success green or a different one?

ahmedqarmout2 replied:
Same

ahmedqarmout2 replied:
Green for showing a question cuz that's a success sign, and there is nothing wrong with showing a question. 
Yellow for hiding the questions as this is a warning for the prof saying hey watch out students won't see this question 
Makes sense?

Mohammed-JS-Khan replied:
:+1: Si senor. Arigato :)

ahmedqarmout2 commented on public/js/admin.js:
```diff
@@ -1,7 +1,7 @@
-    SUCCESS_GREEN: '#4BB543',
-    FAIL_RED: '#D93232',
+    SUCCESS_GREEN: 'green',
+    FAIL_RED: 'red darken-4',
```
Can we add a warning_yellow, might be useful in the future. For example we should warn the prof that the question is now hidden.

ahmedqarmout2 replied:
Drop the colors at the end of the names. Success_green > success and so on

Mohammed-JS-Khan replied:
Done, gonna re-push.

ahmedqarmout2 commented:
Nice work fam. Just a small request to add a warning color, and warn the prof that a question is now hidden. It would be cool to use green for showing and warning for hiding

ahmedqarmout2 reviewed (approved):
Approving but please add warning and use it for hiding

ahmedqarmout2 commented:
Also add the dismiss button plz

mehtadarshan1 reviewed (commented):
Looks good to me

Mohammed-JS-Khan committed:
Added all the edits for my snackbar PR #68

Mohammed-JS-Khan commented:
Added all the required changes and cleaned up the old snackbar stuff.

![image](https://user-images.githubusercontent.com/9061203/31804420-424d7fe4-b527-11e7-9550-c4e9da5373f3.png)

Mohamed-A-Mohamed reviewed (approved):
Nice work. Looks great :+1:

esteeroramy commented on public/js/admin.js:
```diff
@@ -436,7 +437,10 @@ var updateVisibility = function(qid) {
         success: function(data) {
             displayQuestionTable();
-            dropSnack(colours.SUCCESS_GREEN, 'Question ' + qid + ' is now visible to the students');
+            var msg = question['visible'] ? ' is now visible to the students' : ' is now&nbsp<u><b>not</b></u>&nbspvisible to the students';
```
any reason why we are using var over const?

Mohammed-JS-Khan replied:
done.

esteeroramy commented on public/js/admin.js:
```diff
@@ -660,9 +664,10 @@ var sortAccountsTable = function(type) {
-    x.innerHTML=msg;
-    x.className = "show";
-    setTimeout(function(){ x.className = x.className.replace("show", ""); }, 3000);
+    // runs the toast function for 3s with given msg and colour
```
don't you mean 5s?

Mohammed-JS-Khan replied:
good find.

Mohammed-JS-Khan committed:
Changed some typos, vars to consts, and color names.

Mohammed-JS-Khan commented:
@esteeroramy @ahmedqarmout2  completed your changes. Please review.

Mohammed-JS-Khan commented on public/js/admin.js:
```diff
@@ -660,9 +664,10 @@ var sortAccountsTable = function(type) {
+    Materialize.toast(msg + '&nbsp&nbsp&nbsp<i id=closeSnack class="material-icons">close</i>', 5000, 'rounded ' + colour);
 }
+
+$(document).on('click', '#closeSnack', function() {
```
Plz comment on this.

Mohammed-JS-Khan committed:
Added a missing comment.

Mohammed-JS-Khan commented on public/js/admin.js:
```diff
@@ -660,9 +664,11 @@ var sortAccountsTable = function(type) {
+    Materialize.toast(msg + '&nbsp&nbsp&nbsp<i id=closeSnack class="material-icons">close</i>', 5000, 'rounded ' + colour);
 }
+
+/* Listner for the "x" on the snackbar/toasts */
```
Nice english.

Mohammed-JS-Khan committed:
Fixed typo and formatting

Mohammed-JS-Khan committed:
Pulled updates from upstream develop and merged with snackbar changes
